addons/mail/models/mail_message.py

```diff
@@ -414,8 +413,18 @@ def message_fetch_failed(self):
         return messages._format_mail_failures()
 
     @api.model
-    def message_fetch(self, domain, limit=20):
-        return self.search(domain, limit=limit).message_format()
+    def message_fetch(self, domain, moderated_channels=None, limit=20):
+        messages = self.search(domain, limit=limit)
+        if moderated_channels and self.env.user.moderation_channel_ids:
```

Comment:
If we want to speedup a bit shouldn't we check that moderated_channels is effectively in self.env.user.moderation_channel_ids.ids ? If user is not a moderator of the target chanenls no need to perform an additional check if I am not wrong.